server/CGameHandler.cpp

```diff
@@ -5209,16 +5209,23 @@ void CGameHandler::attackCasting(const BattleAttack & bat, Bonus::BonusType atta
 	}
 }
 
-void CGameHandler::handleAttackBeforeCasting (const BattleAttack & bat)
+void CGameHandler::handleAttackBeforeCasting(BattleAttack *bat)
 {
-	const CStack * attacker = gs->curB->battleGetStackByID(bat.stackAttacking);
-	attackCasting(bat, Bonus::SPELL_BEFORE_ATTACK, attacker); //no death stare / acid breath needed?
+	const CStack * attacker = gs->curB->battleGetStackByID(bat->stackAttacking);
+	attackCasting(*bat, Bonus::SPELL_BEFORE_ATTACK, attacker); //no death stare / acid breath needed?
+	// filter possibly dead stacks
+	bat->bsa.erase(std::remove_if(bat->bsa.begin(), bat->bsa.end(),
+	               [this](const BattleStackAttacked &bsa)
+	               {
+	                 return gs->curB->getStack(bsa.stackAttacked) == nullptr;
```

Comment:
Use const accessor battleGetStackByID in gamehandler. getStack should be used only in applyGs